tests/common.py

```diff
@@ -0,0 +1,133 @@
+    with open(plan_path, "r") as f:
+        return f.read()
+
+
+class SubstraitUtils:
+    """
+    Common utility for substrait integration tests.
+    """
+
+    @staticmethod
+    def get_full_path(file_names: Iterable[str]) -> list[str]:
+        """
+        Get full paths for the TPCH parquet data.
+
+        Parameters:
+            file_names:
+                List of TPCH parquet data file names provided by the test case.
+
+        Returns:
+            List of full paths.
+        """
+        data_dir = CUR_DIR / "data" / "tpch_parquet"
+        full_paths_list = [f"{data_dir}/{dataset}" for dataset in file_names]
+
+        return full_paths_list
+
+    @staticmethod
+    def arrow_sort_tb_values(table: pa.Table, sortby: Iterable[str]) -> pa.Table:
+        """
+        Sort the pyarrow table by the given list of columns.
+
+        Parameters:
+            table:
+                Original pyarrow Table.
+            sortby:
+                Columns to sort the results by.
+
+        Returns:
+            Pyarrow Table sorted by given columns.
+
+        """
+        table_sorted_indexes = pa.compute.bottom_k_unstable(
+            table, sort_keys=sortby, k=len(table)
+        )
+        table_sorted = table.take(table_sorted_indexes)
+        return table_sorted
+
+    def format_sql_query(self, sql_query: str, file_names: list[str]) -> str:
+        """
+        Replace the 'Table' Parameters from the SQL query with the relative
+        file paths of the parquet data.
+
+        Parameters:
+            sql_query:
+                SQL query.
+            file_names:
+                List of file names.
+
+        Returns:
+            SQL Query with file paths.
+        """
+        sql_commands_list = [line.strip() for line in sql_query.strip().split("\n")]
+        sql_query = " ".join(sql_commands_list)
+        # Get full path for all datasets used in the query
+        parquet_file_paths = self.get_full_path(file_names)
+
+        return sql_query.format(*parquet_file_paths)
+
+    def format_substrait_query(
+        self, substrait_query: str, file_names: list[str]
+    ) -> bytes:
+        """
+        Replace the 'local_files' path in the substrait query plan with
+        the full path of the parquet data.
+
+        Parameters:
+            substrait_query:
+                Substrait query.
+            file_names:
+                List of file names.
+
+        Returns:
+            Substrait query plan in byte format.
+        """
+        # Get full path for all datasets used in the query
+        parquet_file_paths = self.get_full_path(file_names)
+
+        # Replace the filename placeholder in the substrait query plan with
+        # the proper parquet data file paths.
+        for count, file_path in enumerate(parquet_file_paths):
+            substrait_query = substrait_query.replace(
+                f"FILENAME_PLACEHOLDER_{count}", file_path
+            )
+
+        return pa.lib.tobytes(substrait_query)
```

Comment:
```suggestion
        return substrait_query.encode()
```
We should probably try and avoid pyarrow dependencies outside of the consumer.  Also, is there a particular reason this is needed?  Any JSON should be representable as `UTF-8` so I think it's safe, mostly just curious.  If it's Acero that is requiring this then you might do the encoding in `AceroConsumer` just to keep things simpler.

Reply:
Yeah, Acero is requiring it to be bytes.  I'll do the encoding in AceroConsumer.